test/tests/test_sync_manage.js

```diff
@@ -1,6 +1,6 @@
 var process = require("process");
 
-var syncClient = require("../../src/sync-client");
+var syncClient = require("../../src/sync-client")();
```

Comment:
I have done some research where is that used. This change will cause a little bit of headache as it will not be backwards compatible. I think this change is solid and we should merge however with this format it can cause problems with multiple sample apps, templates, performance tests, openshift starters etc.

In my opinion it's easy to have something like this:
```
require("../../src/sync-client"); // Works as usual 
// provides factory method (with single instance always available globally
require("../../src/sync-client").factory(); 
```

Then we can merge that without worrying about backwards compatibility and breaking other clients.

Reply:
@wtrocki just curious where is the `sync-client.js` is being used directly, instead of using `index.js` file?

Reply:
Actually factory is there and everything works from the js side. I had problems with TypeScript but this was resolved already.